### PR TITLE
test(link-preview): refactor for cypress tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

1. Refactor Cypress tests for the `Link Preview` component to use new cypress-component-react framework for testing.
Add `link-preview.test.js`. 
2. Implemented new function for checking `goldenOutline` in common steps using jQuery.

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `link-preview.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

File: src/components/link-preview/link-preview.test.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  });
+
+  describe("check events for Link Preview component", () => {
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onClick callback when a click event is triggered", () => {
```
Comment (stephenogorman) [inline src/components/link-preview/link-preview.test.js:130]: Should this read "should call onClose callback..."?

File: src/components/link-preview/link-preview.test.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      linkPreviewCloseIcon()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it.each([["Enter"], ["Space"]])(
+      "should call onClick callback when a keyboard event is triggered",
```
Comment (stephenogorman) [inline src/components/link-preview/link-preview.test.js:144]: Same as previous comment.

**Review by stephenogorman — CHANGES_REQUESTED** [review]

File: src/components/link-preview/link-preview.test.js (hunk 2 above)
Comment (DlgSHi) [reply src/components/link-preview/link-preview.test.js]: fixed

**Review by stephenogorman — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**